Marlin/src/gcode/eeprom/M500-M504.cpp

```diff
@@ -73,7 +73,7 @@ void GcodeSuite::M502() {
           bool success = true;
           for (uint16_t i = 0; success && i < sizeof(mc_zip); ++i) {
             const uint8_t c = pgm_read_byte(&mc_zip[i]);
-            file.write(c);
+            success = (file.write(&c) == 1);
```

Comment:
This sure looks like it would call…
```cpp
void SdFile::write(const char * const str) { SdBaseFile::write(str, strlen(str)); }
```
…which writes all bytes at the given string address up to the first `nul`.

Reply:
Not really because mc_zip is a binary array and probably contains more than one 0x00

Reply:
I fixed it and it actually compiles now, and I did a little `SDFile` cleanup besides. Thanks for the improvement!